### PR TITLE
vendor: update system-validators to v1.3.0

### LINKED ISSUE #2335: Check for /sys/fs/cgroup/cpu/cpu.cfs_quota_us / config_cfs_bandwith on init
## What keywords did you search in kubeadm issues before filing this one?

cfs_quota_us aarch64 centos8 cri-o crio

```
Oct 24 01:00:44 chaos kubelet[14438]: E1024 01:00:44.465714   14438 kuberuntime_manager.go:804] container &Container{Name:etcd,Image:k8s.gcr.io/etcd:3.4.13-0,Command:[etcd --advertise-client-urls=https://192.168.1.70:2379 --cert-file=/et>
Oct 24 01:00:44 chaos kubelet[14438]: time="2020-10-24T01:00:44Z" level=error msg="container_linux.go:349: starting container process caused \"process_linux.go:449: container init caused \\\"process_linux.go:415: setting cgroup config fo>
Oct 24 01:00:44 chaos kubelet[14438]: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:415: setting cgroup config for procHooks process caused \\\"failed to write \>
Oct 24 01:00:44 chaos kubelet[14438]: E1024 01:00:44.465863   14438 pod_workers.go:191] Error syncing pod e0f6df37808b0e77cc354267a4dc6b38 ("etcd-chaos_kube-system(e0f6df37808b0e77cc354267a4dc6b38)"), skipping: failed to "StartContainer">
```

## Is this a BUG REPORT or FEATURE REQUEST?

FEATURE REQUEST

As seen in:
https://github.com/cri-o/cri-o/issues/4307
https://bugs.centos.org/view.php?id=17813

kubeadm should check for the existance of /sys/fs/cgroup/cpu/cpu.cfs_quota_us.  It missing is a sign of the kernel being compiled without ```config_cfs_bandwith``` which results in the errors above and a failed kubeadm init.

## Versions

kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:47:53Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/arm64"}

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:

Includes the following changes to kernel validation:
- add required options: CGROUP_PIDS, FAIR_GROUP_SCHED
- add optional options: CFS_BANDWIDTH, CGROUP_HUGETLB

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubeadm/issues/2335

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
kubeadm: amend the node kernel validation to treat CGROUP_PIDS, FAIR_GROUP_SCHED as required and CFS_BANDWIDTH, CGROUP_HUGETLB as optional
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: vendor/k8s.io/system-validators/validators/types_unix.go [hunk 1]
```diff
@@ -48,11 +49,14 @@ var DefaultSysSpec = SysSpec{
 			{Name: "PROC_FS"},
 			{Name: "NETFILTER_XT_TARGET_REDIRECT", Aliases: []string{"IP_NF_TARGET_REDIRECT"}},
 			{Name: "NETFILTER_XT_MATCH_COMMENT"},
+			{Name: "FAIR_GROUP_SCHED"},
```
Comment (neolit123) [inline vendor/k8s.io/system-validators/validators/types_unix.go:52]: WRT the new required fields:
https://github.com/kubernetes/kubeadm/issues/2335

@neolit123 

> both these PRs introduce new required configuration options (FAIR_GROUP_SCHED and CGROUP_PIDS), which can trip existing setups:
kubernetes/system-validators#19
kubernetes/system-validators#20
how common are these options and do you have any concerns about that?
i'm less concerned about CGROUP_PIDS, for some reason.

@odinuge 

> It has never been supported (or been working) running with a configuration that fails with the new validation, so I think it should be good to go. The only places they might be disabled, from my experience, are raspberry-pi like devices used for prototyping.

**Comment by neolit123** [comment]
waiting to see if our CI passes with the new validation.

**Comment by neolit123** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/96378/pull-kubernetes-bazel-test/1325833508236562432
> go_test: //pkg/kubelet/volumemanager/reconciler/go_default_test:run_1_of_2

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/96378/pull-kubernetes-integration/1325833508295282688
> k8s.io/kubernetes/test/integration/dryrun: TestDryRun

flakes
/retest

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Review by odinuge — APPROVED** [review]
Nice! 
/lgtm

**Comment by dims** [comment]
@neolit123 looks like this needs a rebase. please assign to liggitt after that.

**Comment by neolit123** [comment]
> @neolit123 looks like this needs a rebase. please assign to liggitt after that.

rebased.
/assign @liggitt

File: vendor/k8s.io/system-validators/validators/types_unix.go (hunk 1 above)
Comment (liggitt) [reply vendor/k8s.io/system-validators/validators/types_unix.go:52]: can you clarify what the implications of "both these PRs introduce new required configuration options" are? does this mean an existing configuration will no longer work?

File: vendor/k8s.io/system-validators/validators/types_unix.go [hunk 2]
```diff
@@ -48,11 +49,14 @@ var DefaultSysSpec = SysSpec{
...
 		},
 		Optional: []KernelConfig{
 			{Name: "OVERLAY_FS", Aliases: []string{"OVERLAYFS_FS"}, Description: "Required for overlayfs."},
 			{Name: "AUFS_FS", Description: "Required for aufs."},
 			{Name: "BLK_DEV_DM", Description: "Required for devicemapper."},
+			{Name: "CFS_BANDWIDTH", Description: "Required for CPU quota."},
+			{Name: "CGROUP_HUGETLB", Description: "Required for hugetlb cgroup."},
```
Comment (liggitt) [inline vendor/k8s.io/system-validators/validators/types_unix.go:59]: what does "Required for ..." mean in an `Optional` list?

File: vendor/k8s.io/system-validators/validators/types_unix.go (hunk 1 above)
Comment (neolit123) [reply vendor/k8s.io/system-validators/validators/types_unix.go:52]: @odinuge confirmed that kernels without these options (now required by the library) never worked with the kubelet.

**Comment by liggitt** [comment]
/lgtm
/approve
dependencies look good

/hold is this fixing a blocker/critical bug for 1.20 (it's flagged as a feature)? if not, let's move to the 1.21 milestone

File: vendor/k8s.io/system-validators/validators/types_unix.go (hunk 2 above)
Comment (neolit123) [reply vendor/k8s.io/system-validators/validators/types_unix.go:59]: can be interpreted as required kernel options for optional k8s features. no preference on the wording in the description on my side.
cc @KentaTada

**Comment by neolit123** [comment]
> /hold is this fixing a blocker/critical bug for 1.20 (it's flagged as a feature)? if not, let's move to the 1.21 milestone

a validating feature that didn't merge before code-freeze.
punting to 1.21 seems fine.

/milestone v1.21

**Pull request merged**